platform/src/test/java/org/exoplatform/platform/qa/ui/platform/social/SOCNotificationIntranetViewAllTestIT.java

```diff
@@ -0,0 +1,364 @@
+    executeJavaScript("window.scrollBy(0,-1500)", "");
+    info("user 2 likes user1's activity");
+    refresh();
+    executeJavaScript("window.scrollBy(0,-5500)", "");
+    homePagePlatform.goToHomePage();
+    activityStream.likeActivity(activity1);
+
+    info("user2 mentions user1 in activity stream");
+    homePagePlatform.goToHomePage();
+    activityStream.mentionUserActivity(username1, activity4);
+    activityStream.checkActivity(activity4);
+
+    info("user 2 posts activity in user1's activity stream");
+    homePagePlatform.goToConnections();
+    $(byText(username1 + " " + username1)).click();
+    navigationToolbar.goToMyActivities();
+    activityStream.addActivity(activity2, "");
+    activityStream.checkActivity(activity2);
+
+    info("User 2 posts activity in activity stream of user1's space");
+    homePagePlatform.goToAllSpace();
+    spaceManagement.acceptAInvitation(space1);
+
+    info("user posted an activity in space");
+    activityStream.addActivity(activity3, "");
+    activityStream.checkActivity(activity3);
+
+    info("Sign in as user 3 and create a space, invite user1 to join space and connect to user1");
+    manageLogInOut.signIn(username3, password);
+    info("User 3 create a space");
+    homePagePlatform.goToAllSpace();
+    spaceManagement.goToCreateSpace();
+    spaceManagement.addNewSpaceSimple(space2, space2);
+    info("User3 invites user1 to join space");
+    spaceHomePage.goToSpaceSettingTab();
+    spaceSettingManagement.inviteUser(username1, false, "");
+    info("user3 connects to user 1");
+    homePagePlatform.goToConnections();
+    connectionsManagement.connectToAUser(username1);
+
+    info("Sign in as user 4 and request to join user1's space");
+    manageLogInOut.signIn(username4, password);
+    homePagePlatform.goToAllSpace();
+    spaceManagement.requestToJoinSpace(space1);
+
+    /*
+     * Step Number: 1 Step Name: Step 1: Open View All page Step Description: -
+     * Login - Click the notifications icon in the top navigation - Click View All
+     * button Input Data: Expected Outcome: - The user is redirected to the View All
+     * page
+     */
+    manageLogInOut.signIn(username1, password);
+    for (int i = 0; i <= 100; i++) {
+      if ($(byId("UIIntranetNotificationsPortlet")).is(Condition.not(Condition.exist))) {
+        navigationToolbar.goToIntranetNotification();
+        intranetNotification.goToAllNotification();
+      } else {
+        break;
+      }
+    }
+    /*
+     * Step Number: 2 Step Name: Step 2: Check Connection Request notification Step
+     * Description: - Click on the Connection Request notification Input Data:
+     * Expected Outcome: - The user is redirect to the profile of User A
+     */
+    $(byText(username3 + " " + username3)).parent()
+                                          .shouldHave(Condition.text(username3 + " " + username3 + " wants to connect with you"));
+    for (int i = 0; i <= 100; i++) {
```

Comment:
Is this loop here to wait for the text to appear ? Cannot we wait in better way ?
Same remark for others loops.